test/BatchInvoker.t.sol

```diff
@@ -18,86 +28,181 @@ contract BatchInvokerTest is Test {
     Callee public callee;
     BatchInvoker public invoker;
     VmSafe.Wallet public authority;
+    VmSafe.Wallet public recipient;
 
     uint8 AUTHCALL_IDENTIFIER = 2;
 
-    uint256 nonce = 0;
-    uint256 value = 0;
-
     function setUp() public {
         invoker = new BatchInvoker();
         callee = new Callee();
         authority = vm.createWallet("authority");
+        recipient = vm.createWallet("recipient");
         vm.label(address(invoker), "invoker");
         vm.label(address(callee), "callee");
         vm.label(authority.addr, "authority");
     }
 
-    function constructAndSignTransaction()
-        internal
-        view
-        returns (uint8 v, bytes32 r, bytes32 s, bytes memory execData)
-    {
-        bytes memory data = abi.encodeWithSelector(Callee.expectSender.selector, address(authority.addr));
-        bytes memory transactions = abi.encodePacked(AUTHCALL_IDENTIFIER, address(callee), value, data.length, data);
-        execData = abi.encode(nonce, transactions);
-        // construct batch digest & sign
-        bytes32 digest = invoker.getDigest(execData);
-        (v, r, s) = vm.sign(authority.privateKey, digest);
+    function test_execute_withData() external {
+        vm.pauseGasMetering();
+        uint256 nonce = invoker.nextNonce(authority.addr);
+
+        bytes memory data = abi.encodeWithSelector(Callee.increment.selector);
+        bytes memory calls;
+        calls = abi.encodePacked(AUTHCALL_IDENTIFIER, address(callee), uint256(0), data.length, data);
```

Comment:
nit: i'd make a util to encode calls, usable in tests or scripts, also helps demonstrate the encoding